sdk/couchbase-core/src/querycomponent.rs

```diff
@@ -85,10 +91,13 @@ impl<C: Client> QueryComponent<C> {
     }
 
     pub async fn query(&self, opts: QueryOptions) -> error::Result<QueryResultStream> {
-        let retry_info = RetryInfo::new(
-            opts.read_only.unwrap_or_default(),
-            opts.retry_strategy.clone(),
-        );
+        let retry = if let Some(retry_strategy) = opts.retry_strategy.clone() {
+            retry_strategy
+        } else {
+            DEFAULT_RETRY_STRATEGY.clone()
+        };
```

Comment:
nit: 
Could do:
```
let retry = opts.retry_strategy.clone().unwrap_or( DEFAULT_RETRY_STRATEGY.clone())
```

Reply:
I didn't want to end up doing 2 clones in the case of falling back to the default.